File: apps/cargo-tracker/src/CargoTracker.Web/Routing/Application/Internal/CommandServices/RegisterVoyageCommand.cs
using CargoTracker.Routing.Domain.Model;

namespace CargoTracker.Routing.Application.Internal.CommandServices;

public sealed record RegisterVoyageCommand(
    string VoyageNumber,
    string VesselName,
    string Carrier,
    IReadOnlyList<SupportedCargoType> SupportedCargoTypes,
    IReadOnlyList<RegisterCarrierMovementCommand> CarrierMovements);

public sealed record RegisterCarrierMovementCommand(
    string DepartureLocationUnLocode,
    string ArrivalLocationUnLocode,
    DateTimeOffset DepartureDate,
    DateTimeOffset ArrivalDate,
    int SequenceNumber);

File: apps/backend/routingms/src/main/java/com/example/routingms/application/internal/RegisterVoyageCommand.java
package com.example.routingms.application.internal;

import com.example.routingms.domain.model.CargoType;
import com.example.routingms.domain.model.Schedule;
import com.example.routingms.domain.model.VoyageNumber;
import java.util.Set;

/** 航海スケジュールの登録・更新の入力（US24・US25）。 */
public record RegisterVoyageCommand(
        VoyageNumber voyageNumber,
        String vesselName,
        String carrierName,
        Set<CargoType> supportedCargoTypes,
        Schedule schedule) {
}
